src/utils.ts

```diff
@@ -0,0 +1,99 @@
+import * as core from '@actions/core'
+import { CITelemetryData, JobInfo, MetaData, ProcessTelemetryDatum, TelemetryDatum } from './interfaces';
+import * as github from '@actions/github';
+import axios from 'axios';
+import * as path from 'path';
+
+export const WORKFLOW_TELEMETRY_SERVER_PORT = "WORKFLOW_TELEMETRY_SERVER_PORT";
+
+export const WORKFLOW_TELEMETRY_VERSIONS = {
+    METRIC: "v1",
+    PROCESS: "v1"
+};
+
+const WORKFLOW_TELEMETRY_BASE_URL = `${process.env["WORKFLOW_TELEMETRY_BASE_URL"] || "https://foresight.service.thundra.io"}`
+
+export const WORKFLOW_TELEMETRY_ENDPOINTS = {
+    METRIC: new URL(path.join("/api", WORKFLOW_TELEMETRY_VERSIONS.METRIC, "/telemetry/metrics"), WORKFLOW_TELEMETRY_BASE_URL).toString(),
+    PROCESS: new URL(path.join("/api", WORKFLOW_TELEMETRY_VERSIONS.METRIC, "/telemetry/processes"), WORKFLOW_TELEMETRY_BASE_URL).toString()
+};
+
+export const JOB_STATES_NAME = {
+    FORESIGHT_WORKFLOW_JOB_ID: "FORESIGHT_WORKFLOW_JOB_ID",
+    FORESIGHT_WORKFLOW_JOB_NAME: "FORESIGHT_WORKFLOW_JOB_NAME",
+    FORESIGHT_WORKFLOW_JOB_RUN_ATTEMPT: "FORESIGHT_WORKFLOW_JOB_RUN_ATTEMPT"
+}
+
+export async function setServerPort() {
+    var portfinder = require('portfinder');
+    portfinder.basePort = 10000;
+    const port = parseInt(process.env.WORKFLOW_TELEMETRY_SERVER_PORT || '');
+    if(!port) {
+        process.env["WORKFLOW_TELEMETRY_SERVER_PORT"] = await portfinder.getPortPromise();
+    }
+    core.saveState(WORKFLOW_TELEMETRY_SERVER_PORT, process.env.WORKFLOW_TELEMETRY_SERVER_PORT);
+    logger.info(`Workflow telemetry server port is: ${process.env.WORKFLOW_TELEMETRY_SERVER_PORT}`);
+}
+
+
+export function saveJobInfos(jobInfo: JobInfo) {
+    core.exportVariable(JOB_STATES_NAME.FORESIGHT_WORKFLOW_JOB_ID, jobInfo.id)
+    core.exportVariable(JOB_STATES_NAME.FORESIGHT_WORKFLOW_JOB_NAME, jobInfo.name)
+}
+
+function getJobInfo(): JobInfo {
+    const jobInfo: JobInfo = {
+        id: parseInt(process.env[JOB_STATES_NAME.FORESIGHT_WORKFLOW_JOB_ID] || ''),
+        name: process.env[JOB_STATES_NAME.FORESIGHT_WORKFLOW_JOB_NAME],
+    }
+    return jobInfo
+}
+
+function getMetaData(): MetaData {
+    const { repo, runId } = github.context
+    const jobInfo = getJobInfo();
+    const metaData: MetaData = {
+        ciProvider: "GITHUB",
+        runId: runId,
+        repoName: repo.repo,
+        repoOwner: repo.owner,
+        runAttempt: process.env.GITHUB_RUN_ATTEMPT,
+        runnerName: process.env.RUNNER_NAME,
+        jobId: jobInfo.id,
+        jobName: jobInfo.name,
+    }
+    return metaData
+}
+
+export function createCITelemetryData(telemetryData: ProcessTelemetryDatum): CITelemetryData {
+    return {
+        metaData: getMetaData(),
+        telemetryData: telemetryData
+    }
+}
+
+export async function sendData (url :string, ciTelemetryData: CITelemetryData)
+{
+    logger.debug(`Send data url: ${url} ${core.getInput("api_key")}`);
+    try {
+        const { data } = await axios.post(
```

Comment:
Better to check response status code and log error if status code is not `200`

Reply:
Axios automatically throws error w.r.t status code except successful responses and we check it in catch scope with axios.isAxiosError(error).